packages/commonwealth/test/integration/api/cosmosCache.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { sleep } from '../../util/delayUtils';
+
+function verifyNoCacheResponse(res) {
+  expect(res.body).to.not.be.null;
+  expect(res).to.have.status(200);
+  expect(res).to.not.have.header('X-Cache', 'HIT');
+}
+
+async function verifyCacheResponse(key, res, resEarlier) {
+  expect(res).to.have.status(200);
+  expect(res).to.have.header('X-Cache', 'HIT');
+  const valFromRedis = await cacheDecorator.checkCache(key);
+  expect(valFromRedis).to.not.be.null;
+  expect(JSON.parse(valFromRedis)).to.be.deep.equal(res.body);
+  expect(JSON.parse(valFromRedis)).to.be.deep.equal(resEarlier.body);
+}
+
+describe('Cosmos Cache', () => {
+  const redisCache: RedisCache = new RedisCache();
+  const route_namespace: RedisNamespaces = RedisNamespaces.Route_Response;
+
+  before(async () => {
+    await resetDatabase();
+    await connectToRedis(redisCache);
+    cacheDecorator.setCache(redisCache);
+  });
+
+  after(async () => {
+    await redisCache.deleteNamespaceKeys(route_namespace);
+    await redisCache.closeClient();
+  });
+
+  describe('Cosmos Cache', () => {
+    describe('cosmosAPI', () => {
+      async function makeRPCRequest(
+        body,
+        path = '/cosmosAPI/juno',
+        headers = {
+          'content-type': 'text/plain;charset=UTF-8',
+          'accept-language': 'en-US,en;q=0.9',
+        }
+      ) {
+        return chai.request(app).post(path).set(headers).send(body);
+      }
+
+      async function rpcTestBody(body, key) {
+        const res1 = await makeRPCRequest(body);
+        verifyNoCacheResponse(res1);
+        const res2 = await makeRPCRequest(body);
+        await verifyCacheResponse(key, res2, res1);
+      }
+
+      const rpcProposalsCacheExpectedTest = async (proposalStatus: string) => {
+        const pathParam = `{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"${proposalStatus}","prove":false}`;
+        const body = `{"jsonrpc":"2.0","id":382288611593,"method":"abci_query","params":${pathParam}}`;
+        const key = `/cosmosAPI/juno_{"path":"/cosmos.gov.v1beta1.Query/Proposals","data":"${proposalStatus}","prove":false}`;
+        await rpcTestBody(body, key);
+      };
+
+      it('should cache passed proposals', async () => {
+        await rpcProposalsCacheExpectedTest('0803');
+      });
+      it('should cache passed proposals (0803220a0a080000000000000087)', async () => {
+        await rpcProposalsCacheExpectedTest('0803220a0a080000000000000087');
+      });
+      it('should cache passed proposals (0803220a0a080000000000000100)', async () => {
+        await rpcProposalsCacheExpectedTest('0803220a0a080000000000000100');
+      });
+      it('should cache passed proposals (0803220a0a080000000000000189)', async () => {
+        await rpcProposalsCacheExpectedTest('0803220a0a080000000000000189');
+      });
+      it('should cache rejected proposals', async () => {
+        await rpcProposalsCacheExpectedTest('0804');
+      });
+      it('should cache rejected proposals', async () => {
```

Comment:
repeated description "should cache rejected proposals"

Reply:
good catch. should be "failed"

Reply:
fixed